### DISCUSSION TITLE
StringCompress

### DISCUSSION THREAD

**Discussion by le-rou** [VentuzTechnology/Ventuz.Extension.prerelease#11] [Show and tell]
As a first tryout I wrote string compress and decompress content nodes. I needed this in one of my last projects. We sent some data as JSON via OSC. Unfortunately, the sent string got a bit too long, so I just compressed it. Now I put this into nodes to see how everything works. I used an enum as an input to select the compression method.

**Comment by VentuzRalfStanke** [comment]
Hej Rou, 

you could do both (compresssion and decompression) in a single node, be extending the enum for all oparations, because the signature of the nodes is the same for both: string+comptype in, string out. The node could change its icon for each operation type.

Another idea is not to use base64 encoding, Ventuz supports byte arrays...

... just ideas,

Best regards,
Rafl
